### PR TITLE
Session details

### PR DESCRIPTION
**Pull request by TikhomirovSergey**
## Change list
- a new interface IHasSessionDetails
- the  IHasSessionDetails is implemented by AppiumDriver
- tests
 
## Types of changes
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Details
I have started to add features which have been add to java clint since Jan'17

### CODE REVIEW

File: appium-dotnet-driver/Appium/AppiumDriver.cs [hunk 1]
```diff
@@ -773,27 +580,61 @@ public void Zoom(IWebElement el)
...
+        {
+            get
+            {
+                var automation = GetSessionDetail("automationName");
+                return automation == null ? null : Convert.ToString(automation);
+            }
+        }
+
+        public bool IsBrowser => GetSessionDetail("browserName") != null
+                                 && Context.IndexOf("NATIVE_APP", StringComparison.OrdinalIgnoreCase) < 0;
```
Comment (Astro03) [inline appium-dotnet-driver/Appium/AppiumDriver.cs]: for both "browerName" and "NATIVE_APP"  can use their "static readonly" in their respective files.

File: appium-dotnet-driver/Appium/AppiumDriver.cs [hunk 2]
```diff
@@ -425,7 +244,7 @@ public void Rotate(Dictionary<string, int> opts)
 
         public void HideKeyboard() => AppiumCommandExecutionHelper.HideKeyboard(this, null, null);
 
-        /// <sumary>
+        /// <summary>
```
[the remark this answers is not in the document]
Comment (TikhomirovSergey) [reply appium-dotnet-driver/Appium/AppiumDriver.cs]: It is necessary here because it is not declared by any interface or super class

**Comment by Astro03** [comment]
@TikhomirovSergey are you online? want to get on slack?

**Comment by mykola-mokhnach** [comment]
@TikhomirovSergey have you tried to contact Slack's customer support? Perhaps there are other ways of getting the code (like Google Authenticator). There is one important topic I'd like to discuss about W3C stuff support by java client

File: appium-dotnet-driver/Appium/AppiumDriver.cs [hunk 3]
```diff
@@ -773,27 +580,43 @@ public void Zoom(IWebElement el)
...
+            return details.ContainsKey(detail) ? details[detail] : null;
+        }
+
+        public string PlatformName => GetSessionDetail(MobileCapabilityType.PlatformName) as string 
+                                      ?? GetSessionDetail(CapabilityType.Platform) as string;
+
+        public string AutomationName => GetSessionDetail(MobileCapabilityType.AutomationName) as string;
+
+        public bool IsBrowser => GetSessionDetail(MobileCapabilityType.BrowserName) != null
+                                 && Context.IndexOf("NATIVE_APP", StringComparison.OrdinalIgnoreCase) < 0;
```
Comment (Astro03) [inline appium-dotnet-driver/Appium/AppiumDriver.cs]: can we use the native_app const as well?

**Comment by Astro03** [comment]
btw, my personal preference is to put {} around all blocks, even if it's a one line if stmt or return.

**Pull request merged**